### ISSUE TITLE
Exception thrown when socket is not yet connected

### ISSUE THREAD

**Issue by SpaceK33z** [CodeYellowBV/spine-high-templar#1]
When you try to send a message while the socket has not connected (because the server is either down or is busy connecting), you get this error:

```
Uncaught DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.
```

It should detect beforehand that the socket is not yet connected and either not send the message, or delay sending it.

**Comment by stam** [comment]
That's already happening, but the default connectDelay apparently isn't long enough.
I could loop the send until the websocket is connected?

**Comment by stam** [comment]
Fixed in v0.2.0
